- '시장가치'와 각 변수들의 상관성을 나타내기 위한 코드입니다.
- 전체 데이터와 각 포지션별 데이터에 대해 진행
- 범주형으로 나타낸 변수들은 제거 후 진행
- 상관성은 피어슨 상관계수를 활용
- 절대값이 0.3 이상인 변수는 어느정도 상관성이 있다고 판단

In [1]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [2]:
import pandas as pd
from scipy import stats

# 전체 데이터에 대해 상관 분석

In [3]:
data_path = '/content/drive/MyDrive/Colab Notebooks/캡스톤_4조/dataset/'
data = pd.read_csv(data_path + 'epl_data.csv')

data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3199 entries, 0 to 3198
Data columns (total 25 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   Unnamed: 0  3199 non-null   int64  
 1   이름          3199 non-null   object 
 2   연봉(€)       3199 non-null   float64
 3   시즌          3199 non-null   int64  
 4   포지션         3199 non-null   object 
 5   클럽팀         3199 non-null   object 
 6   전체 출장 시간    3199 non-null   float64
 7   전체 골        3199 non-null   float64
 8   전체 어시스트     3199 non-null   float64
 9   클린시트        3199 non-null   float64
 10  실점          3199 non-null   float64
 11  경고          3199 non-null   float64
 12  퇴장          3199 non-null   float64
 13  90분당 골 관여   3199 non-null   float64
 14  90분당 어시스트   3199 non-null   float64
 15  90분당 골      3199 non-null   float64
 16  90분당 실점     3199 non-null   float64
 17  카드/90분      3199 non-null   float64
 18  공격수 순위      3199 non-null   float64
 19  미드필더 순위     3199 non-null  

In [4]:
del data['Unnamed: 0'], data['이름'], data['시즌'], data['포지션'], data['클럽팀']
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3199 entries, 0 to 3198
Data columns (total 20 columns):
 #   Column     Non-Null Count  Dtype  
---  ------     --------------  -----  
 0   연봉(€)      3199 non-null   float64
 1   전체 출장 시간   3199 non-null   float64
 2   전체 골       3199 non-null   float64
 3   전체 어시스트    3199 non-null   float64
 4   클린시트       3199 non-null   float64
 5   실점         3199 non-null   float64
 6   경고         3199 non-null   float64
 7   퇴장         3199 non-null   float64
 8   90분당 골 관여  3199 non-null   float64
 9   90분당 어시스트  3199 non-null   float64
 10  90분당 골     3199 non-null   float64
 11  90분당 실점    3199 non-null   float64
 12  카드/90분     3199 non-null   float64
 13  공격수 순위     3199 non-null   float64
 14  미드필더 순위    3199 non-null   float64
 15  수비수 순위     3199 non-null   float64
 16  득점 순위      3199 non-null   float64
 17  나이         3199 non-null   int64  
 18  시장가치       3198 non-null   float64
 19  승점         3199 non-null   int64  
dtypes: float

In [5]:
data_pearson = data.corr(method='pearson')
data_pearson

연봉(€)  전체 출장 시간      전체 골   전체 어시스트      클린시트        실점  \
연봉(€)      1.000000  0.121275  0.317677  0.295904  0.288990 -0.048723   
전체 출장 시간   0.121275  1.000000  0.342923  0.386783  0.724242  0.867834   
전체 골       0.317677  0.342923  1.000000  0.530294  0.400991  0.221254   
전체 어시스트    0.295904  0.386783  0.530294  1.000000  0.451028  0.239416   
클린시트       0.288990  0.724242  0.400991  0.451028  1.000000  0.378903   
실점        -0.048723  0.867834  0.221254  0.239416  0.378903  1.000000   
경고         0.009006  0.547660  0.134822  0.185831  0.344152  0.519696   
퇴장        -0.018374  0.141993  0.033963  0.040978  0.077552  0.138616   
90분당 골 관여  0.333370  0.022531  0.718463  0.615669  0.225933 -0.074355   
90분당 어시스트  0.240445  0.008561  0.301563  0.734331  0.169381 -0.076326   
90분당 골     0.294469  0.026388  0.788222  0.336439  0.195618 -0.048838   
90분당 실점   -0.256975 -0.189998 -0.170445 -0.204844 -0.514418  0.198843   
카드/90분    -0.062978 -0.094045 -0.082082 -0.069930 -0.111712 -0.047355   
공격수 순위    -0.226818  0.139307 -0.571785 -0.273715 -0.025520  0.183380   
미드필더 순위   -0.198666  0.107463 -0.269389 -0.595838 -0.044856  0.162645   
수비수 순위    -0.173776  0.097114 -0.273038 -0.235552 -0.190564  0.266525   
득점 순위     -0.208206 -0.251677 -0.654550 -0.384887 -0.282760 -0.186030   
나이         0.164561  0.076559 -0.044354 -0.070468 -0.012464  0.092186   
시장가치       0.670891  0.193159  0.430822  0.420921  0.362801 -0.005104   
승점         0.473229  0.058370  0.226185  0.269735  0.476085 -0.281400   

                 경고        퇴장  90분당 골 관여  90분당 어시스트    90분당 골   90분당 실점  \
연봉(€)      0.009006 -0.018374   0.333370   0.240445  0.294469 -0.256975   
전체 출장 시간   0.547660  0.141993   0.022531   0.008561  0.026388 -0.189998   
전체 골       0.134822  0.033963   0.718463   0.301563  0.788222 -0.170445   
전체 어시스트    0.185831  0.040978   0.615669   0.734331  0.336439 -0.204844   
클린시트       0.344152  0.077552   0.225933   0.169381  0.195618 -0.514418   
실점         0.519696  0.138616  -0.074355  -0.076326 -0.048838  0.198843   
경고         1.000000  0.304383  -0.046358  -0.030075 -0.042563 -0.054279   
퇴장         0.304383  1.000000  -0.020977  -0.020100 -0.014062 -0.012983   
90분당 골 관여 -0.046358 -0.020977   1.000000   0.731369  0.874109 -0.174613   
90분당 어시스트 -0.030075 -0.020100   0.731369   1.000000  0.308350 -0.143175   
90분당 골    -0.042563 -0.014062   0.874109   0.308350  1.000000 -0.142145   
90분당 실점   -0.054279 -0.012983  -0.174613  -0.143175 -0.142145  1.000000   
카드/90분     0.573211  0.263052  -0.072058  -0.042102 -0.070341  0.092085   
공격수 순위     0.096940  0.032470  -0.649000  -0.296574 -0.694299  0.051601   
미드필더 순위    0.077224  0.032719  -0.567008  -0.702940 -0.290732  0.068077   
수비수 순위     0.016732  0.031797  -0.372891  -0.278251 -0.322652  0.290390   
득점 순위     -0.151792 -0.038341  -0.614603  -0.263203 -0.670226  0.118743   
나이         0.039583  0.032829  -0.107780  -0.109013 -0.072876  0.069790   
시장가치       0.028772 -0.033883   0.407588   0.298423  0.356296 -0.271662   
승점        -0.041887 -0.038561   0.288904   0.242961  0.230391 -0.562567   

             카드/90분    공격수 순위   미드필더 순위    수비수 순위     득점 순위        나이  \
연봉(€)     -0.062978 -0.226818 -0.198666 -0.173776 -0.208206  0.164561   
전체 출장 시간  -0.094045  0.139307  0.107463  0.097114 -0.251677  0.076559   
전체 골      -0.082082 -0.571785 -0.269389 -0.273038 -0.654550 -0.044354   
전체 어시스트   -0.069930 -0.273715 -0.595838 -0.235552 -0.384887 -0.070468   
클린시트      -0.111712 -0.025520 -0.044856 -0.190564 -0.282760 -0.012464   
실점        -0.047355  0.183380  0.162645  0.266525 -0.186030  0.092186   
경고         0.573211  0.096940  0.077224  0.016732 -0.151792  0.039583   
퇴장         0.263052  0.032470  0.032719  0.031797 -0.038341  0.032829   
90분당 골 관여 -0.072058 -0.649000 -0.567008 -0.372891 -0.614603 -0.107780   
90분당 어시스트 -0.042102 -0.296574 -0.702940 -0.278251 -0.263203 -0.109013   
90분당 골    -0.070341 -0.694299 -0.290732 -0.322652 -0.670226 -0.072876   
90분당 실점    0.092085  0.05160

In [ ]:
data_pearson['연봉(€)'].sort_values()

90분당 실점     -0.256975
공격수 순위      -0.226818
득점 순위       -0.208206
미드필더 순위     -0.198666
수비수 순위      -0.173776
카드/90분      -0.062978
실점          -0.048723
퇴장          -0.018374
경고           0.009006
전체 출장 시간     0.121275
나이           0.164561
90분당 어시스트    0.240445
클린시트         0.288990
90분당 골       0.294469
전체 어시스트      0.295904
전체 골         0.317677
90분당 골 관여    0.333370
승점           0.473229
연봉(€)        1.000000
Name: 연봉(€), dtype: float64

포지션마다 변수들의 중요도가 다르므로 전체 데이터에 대한 상관분석은 큰 의미가 없어보인다.

따라서 포지션별로 각각 상관분석을 진행하고자 한다.

# 포지션별 상관분석

In [8]:
# 포지션에 맞는 데이터 불러오는 함수
def load_data(position):
  data = pd.read_csv(data_path + 'epl_{}_data.csv'.format(position))
  del data['Unnamed: 0'], data['이름'], data['시즌'], data['포지션'], data['클럽팀']
  return data

# 연봉과의 상관계수를 반환하는 함수
def corr(data, method):
  data_pearson = data.corr(method=method)
  result = data_pearson['시장가치'].sort_values()
  return result

## 🟡 GK

In [9]:
GK = load_data('GK')
corr(GK, 'pearson')

90분당 실점     -0.337732
나이          -0.313147
수비수 순위      -0.238148
득점 순위       -0.003564
카드/90분       0.114989
90분당 어시스트    0.127744
90분당 골 관여    0.127744
전체 어시스트      0.142104
미드필더 순위      0.159763
실점           0.164193
공격수 순위       0.220793
퇴장           0.253255
경고           0.314918
전체 출장 시간     0.401501
승점           0.492186
클린시트         0.541478
연봉(€)        0.587417
시장가치         1.000000
전체 골              NaN
90분당 골            NaN
Name: 시장가치, dtype: float64

🟡 골키퍼의 시장가치와의 연관성 순위
1. 연봉: 0.587417
2. 클린시트: 0.541478
3. 승점: 0.492186
4. 출장 시간: 0.401501
5. 90분당 실점: -0.337732

## 🔵 DF

In [10]:
DF = load_data('DF')
corr(DF, 'pearson')

수비수 순위      -0.338191
90분당 실점     -0.302603
득점 순위       -0.170021
나이          -0.168989
공격수 순위      -0.160841
미드필더 순위     -0.139078
카드/90분      -0.081821
실점          -0.073300
퇴장          -0.029956
경고           0.000905
전체 출장 시간     0.131008
90분당 어시스트    0.168404
90분당 골       0.179361
전체 골         0.206924
90분당 골 관여    0.237634
전체 어시스트      0.291206
클린시트         0.321392
승점           0.536568
연봉(€)        0.661351
시장가치         1.000000
Name: 시장가치, dtype: float64

🔵 수비수의 시장가치와의 연관성 순위
1. 연봉: 0.661351
2. 승점: 0.536568
3. 수비수 순위: -0.338191
4. 클린시트: 0.321392
5. 90분당 실점: -0.302603

## 🟢 MF

In [11]:
MF = load_data('MF')
corr(MF, 'pearson')

공격수 순위      -0.299915
득점 순위       -0.282011
90분당 실점     -0.260700
미드필더 순위     -0.258348
나이          -0.137098
카드/90분      -0.114303
퇴장          -0.076797
실점          -0.025507
경고           0.001759
전체 출장 시간     0.166448
클린시트         0.320237
90분당 어시스트    0.320268
90분당 골       0.370565
전체 어시스트      0.394566
전체 골         0.416327
90분당 골 관여    0.426893
승점           0.499871
연봉(€)        0.706548
시장가치         1.000000
수비수 순위            NaN
Name: 시장가치, dtype: float64

🟢 미드필더의 시장가치와의 연관성 순위
1. 연봉: 0.706548
2. 승점: 0.499871
3. 90분당 골 관여: 0.426893
4. 전체 골: 0.416327
5. 전체 어시스트: 0.394566

## 🔴 FW

In [12]:
FW = load_data('FW')
corr(FW, 'pearson')

득점 순위       -0.314389
공격수 순위      -0.290104
90분당 실점     -0.263067
미드필더 순위     -0.254407
카드/90분      -0.114815
나이          -0.034106
퇴장           0.015395
경고           0.086729
실점           0.150561
90분당 어시스트    0.240956
전체 출장 시간     0.385978
90분당 골       0.389276
90분당 골 관여    0.432904
클린시트         0.444238
전체 어시스트      0.511088
전체 골         0.526241
승점           0.550588
연봉(€)        0.636852
시장가치         1.000000
수비수 순위            NaN
Name: 시장가치, dtype: float64

🔴 공격수의 시장가치와의 연관성 순위
1. 연봉: 0.636852
2. 승점: 0.550588
3. 전체 골: 0.526241
4. 전체 어시스트: 0.511088
5. 클린시트: 0.444238